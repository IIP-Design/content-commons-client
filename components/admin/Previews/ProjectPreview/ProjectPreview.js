/**
 *
 * PreviewProject
 *
 */

import withModal from 'components/admin/ProjectEdit/withModal/withModal';

const PreviewProject = props => {
  const { modalTrigger, modalContent, options } = props;

  return withModal( props, modalTrigger, modalContent, options );
};

export default PreviewProject;
