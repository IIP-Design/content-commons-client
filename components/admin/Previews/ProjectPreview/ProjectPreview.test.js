import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ProjectPreview from './ProjectPreview';

const Trigger = () => <div className="trigger">Trigger</div>;
const Content = () => <div className="content">Content</div>;

const props = {
  modalTrigger: Trigger,
  modalContent: Content,
  options: {
    one: 'option1',
    two: 'option2'
  }
};

const Component = <ProjectPreview { ...props } />;

describe( '<PreviewProject />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
  } );

  it( 'renders a Modal component', () => {
    const wrapper = shallow( Component );

    expect( wrapper.name() ).toEqual( 'Modal' );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'receives the Trigger component as prop', () => {
    const wrapper = shallow( Component );

    expect( wrapper.prop( 'trigger' ).type() )
      .toEqual( Trigger() );
  } );

  it( 'receives the options as props', () => {
    const wrapper = shallow( Component );

    expect( wrapper.prop( 'one' ) ).toEqual( 'option1' );
    expect( wrapper.prop( 'two' ) ).toEqual( 'option2' );
  } );

  it( 'receives the Content component as child', () => {
    const wrapper = shallow( Component );
    const content = wrapper.find( 'Content' );

    expect( content.type()() ).toEqual( Content() );
  } );
} );
