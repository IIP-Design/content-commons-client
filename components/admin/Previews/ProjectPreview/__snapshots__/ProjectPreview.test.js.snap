// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreviewProject /> renders a Modal component 1`] = `
<Modal
  centered={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  one="option1"
  trigger={<Trigger />}
  two="option2"
>
  <ModalContent>
    <Content />
  </ModalContent>
</Modal>
`;
